Framework/PythonInterface/plugins/algorithms/HB3AAdjustSampleNorm.py

```diff
@@ -0,0 +1,272 @@
+            if has_multiple:
+                if load_files:
+                    out_ws_name = out_ws + "_" + os.path.basename(file).strip(',.nxs')
+                else:
+                    out_ws_name = out_ws + "_" + file
+
+            exp_info = scan.getExperimentInfo(0)
+            self.__move_components(exp_info, height, distance)
+
+            # Get the wavelength from the file sample logs if it exists
+            if exp_info.run().hasProperty("wavelength"):
+                wavelength = exp_info.run().getProperty("wavelength").value
+            elif wavelength == -1:
+                raise RuntimeError("Wavelength not found in sample log and was not provided as input to algorithm")
+
+            # Use ConvertHFIRSCDtoQ (and normalize van), or use ConvertWANDSCtoQ which handles normalization itself
+            if method:
+                if has_van:
+                    van_norm = ReplicateMD(ShapeWorkspace=scan, DataWorkspace=vanws, StoreInADS=False)
+                    van_norm = DivideMD(LHSWorkspace=scan, RHSWorkspace=van_norm, StoreInADS=False)
+                    ConvertHFIRSCDtoMDE(InputWorkspace=van_norm, Wavelength=wavelength, MinValues=minvals,
+                                        MaxValues=maxvals, OutputWorkspace=out_ws_name)
+                else:
+                    ConvertHFIRSCDtoMDE(InputWorkspace=scan, Wavelength=wavelength, MinValues=minvals,
+                                        MaxValues=maxvals, OutputWorkspace=out_ws_name)
+                if merge:
+                    wslist.append(out_ws_name)
+            else:
+                # Convert to Q space and normalize with from the vanadium
+                if has_van:
+                    ConvertWANDSCDtoQ(InputWorkspace=scan, NormalisationWorkspace=vanws, Frame='Q_sample',
+                                      Wavelength=wavelength, NormaliseBy='Monitor', BinningDim0=bin0, BinningDim1=bin1,
+                                      BinningDim2=bin2,
+                                      OutputWorkspace=out_ws_name)
+                else:
+                    ConvertWANDSCDtoQ(InputWorkspace=scan, Frame='Q_sample',
+                                      Wavelength=wavelength, NormaliseBy='Monitor', BinningDim0=bin0, BinningDim1=bin1,
+                                      BinningDim2=bin2,
+                                      OutputWorkspace=out_ws_name)
+
+        if method:
+            if merge and len(wslist) > 1:
+                out_ws_name = out_ws
+                MergeMD(InputWorkspaces=wslist, OutputWorkspace=out_ws_name)
+                DeleteWorkspaces(wslist)
+            if has_van:
+                DeleteWorkspace(van_norm)
+
+        # Don't delete workspaces if they were passed in
+        if load_van:
+            DeleteWorkspace(vanws)
+        if load_files:
+            DeleteWorkspace(scan)
+
+        self.setProperty("OutputWorkspace", out_ws_name)
+
+    def __move_components(self, exp_info, height, distance):
```

Comment:
So the way the instrument definition is set up the height (y) is set in the `bank?` component and the distance (x-z) is set in `bank?/panel` component, so to move it correctly you need to do it by

```python
component = exp_info.componentInfo()
for bank in range(1, 4):
    # Set height offset (y) first on bank?                                                                                                                                                            
    index = component.indexOfAny("bank{}".format(bank))

    if height != 0.0:
        offset = V3D(0, height, 0)
        pos = component.position(index)
        offset += pos
        component.setPosition(index, offset)

    # Set distance offset to detector (x,z) on bank?/panel
    if distance != 0.0:
        panel_index = int(component.children(index)[0]) # should only have one child                                                                                                               
        panel_pos = component.position(panel_index)
        panel_rel_pos = component.relativePosition(panel_index)
        # need to move detector in direction in x-z plane 
        panel_offset = panel_rel_pos*(distance/panel_rel_pos.norm())
        panel_offset += panel_pos
        component.setPosition(panel_index, panel_offset)
```